pandas/_libs/groupby.pyx

```diff
@@ -16,14 +16,20 @@ from numpy cimport (ndarray,
 from libc.stdlib cimport malloc, free
 
 from util cimport numeric, get_nat
-from algos cimport swap
-from algos import take_2d_axis1_float64_float64, groupsort_indexer
+from algos cimport (swap, TiebreakEnumType, TIEBREAK_AVERAGE, TIEBREAK_MIN,
```

Comment:
maybe import directly as the name of the enum?

can you also share this name with the non-grouping rank functions?

Reply:
I tried this back in d09268b1 but all of the Py27 builds failed as a result with the error below (py3 was fine). Tried digging up info but couldn't find anything - any chance you've seen this before?

```bash

cythoning pandas/_libs/groupby.pyx to pandas/_libs/groupby.c

Error compiling Cython file:
------------------------------------------------------------
...

        if keep_na and (values[_as[i], 0] != values[_as[i], 0]):
            grp_na_count += 1
            out[_as[i], 0] = np.nan
        else:
            if tiebreak == TiebreakEnumType.TIEBREAK_AVERAGE:
                                          ^
------------------------------------------------------------

pandas/_libs/groupby.pyx:178:43: Compiler crash in AnalyseExpressionsTransform

ModuleNode.body = StatListNode(groupby.pyx:4:0)
StatListNode.stats[16] = StatListNode(groupby.pyx:125:0)
StatListNode.stats[0] = CompilerDirectivesNode(groupby.pyx:125:0)
CompilerDirectivesNode.body = StatListNode(groupby.pyx:125:0)
StatListNode.stats[0] = DefNode(groupby.pyx:125:0,
    doc = u'\n    Only transforms on axis=0\n    ',
    modifiers = [...]/0,
    name = u'group_rank_object',
    num_required_args = 8,
    py_wrapper_required = True,
    reqd_kw_flags_cname = '0',
    used = True)
File 'Nodes.py', line 430, in analyse_expressions: StatListNode(groupby.pyx:130:4)
File 'Nodes.py', line 430, in analyse_expressions: StatListNode(groupby.pyx:170:4)
File 'Nodes.py', line 6181, in analyse_expressions: ForInStatNode(groupby.pyx:170:4)
File 'Nodes.py', line 430, in analyse_expressions: StatListNode(groupby.pyx:171:8)
File 'Nodes.py', line 5842, in analyse_expressions: IfStatNode(groupby.pyx:174:8)
File 'Nodes.py', line 430, in analyse_expressions: StatListNode(groupby.pyx:178:12)
File 'Nodes.py', line 5840, in analyse_expressions: IfStatNode(groupby.pyx:178:12)
File 'Nodes.py', line 5885, in analyse_expressions: IfClauseNode(groupby.pyx:178:15)
File 'ExprNodes.py', line 541, in analyse_temp_boolean_expression: PrimaryCmpNode(groupby.pyx:178:24,
    operator = u'==',
    result_is_used = True,
    use_managed_ref = True)
File 'ExprNodes.py', line 11893, in analyse_types: PrimaryCmpNode(groupby.pyx:178:24,
    operator = u'==',
    result_is_used = True,
    use_managed_ref = True)
File 'ExprNodes.py', line 6329, in analyse_types: AttributeNode(groupby.pyx:178:43,
    attribute = u'TIEBREAK_AVERAGE',
    initialized_check = True,
    is_attribute = 1,
    needs_none_check = True,
    result_is_used = True,
    use_managed_ref = True)
File 'ExprNodes.py', line 6395, in analyse_as_type_attribute: AttributeNode(groupby.pyx:178:43,
    attribute = u'TIEBREAK_AVERAGE',
    initialized_check = True,
    is_attribute = 1,
    needs_none_check = True,
    result_is_used = True,
    use_managed_ref = True)
File 'ExprNodes.py', line 6438, in as_name_node: AttributeNode(groupby.pyx:178:43,
    attribute = u'TIEBREAK_AVERAGE',
    initialized_check = True,
    is_attribute = 1,
    needs_none_check = True,
    result_is_used = True,
    use_managed_ref = True)
File 'ExprNodes.py', line 1906, in analyse_rvalue_entry: NameNode(groupby.pyx:178:43,
    cf_maybe_null = True,
    is_name = True,
    name = u'TIEBREAK_AVERAGE',
    result_is_used = True,
    use_managed_ref = True)
File 'ExprNodes.py', line 1939, in analyse_entry: NameNode(groupby.pyx:178:43,
    cf_maybe_null = True,
    is_name = True,
    name = u'TIEBREAK_AVERAGE',
    result_is_used = True,
    use_managed_ref = True)
File 'ExprNodes.py', line 1953, in check_identifier_kind: NameNode(groupby.pyx:178:43,
    cf_maybe_null = True,
    is_name = True,
    name = u'TIEBREAK_AVERAGE',
    result_is_used = True,
    use_managed_ref = True)

Compiler crash traceback from this point on:
  File "/Users/williamayd/miniconda3/envs/pandas_dev2/lib/python2.7/site-packages/Cython/Compiler/ExprNodes.py", line 1953, in check_identifier_kind
    if entry.is_type and entry.type.is_extension_type:
AttributeError: 'NoneType' object has no attribute 'is_type'

```

Reply:
thats annoying. havne't seen that